python/kfserving/kfserving/storage.py

```diff
@@ -253,8 +253,10 @@ def _download_from_uri(uri, out_dir=None):
                     .startswith('application/zip'):
                 raise RuntimeError("URI: %s did not respond with \'Content-Type\': \'application/zip\'" % uri)
             if mimetype == 'application/x-tar' and not response.headers.get('Content-Type', '')\
-                    .startswith('application/x-tar'):
-                raise RuntimeError("URI: %s did not respond with \'Content-Type\': \'application/x-tar\'" % uri)
+                    .startswith(('application/x-tar', 'application/x-gtar', 'application/x-gzip', 'application/gzip')):
```

Comment:
prefer using a variable for the content type list so we do not have to repeat in a lot of the places